roles/k3s/post/tasks/templates/metallb.crs.j2

```diff
@@ -5,10 +5,17 @@ metadata:
   namespace: metallb-system
 spec:
   addresses:
-  - {{ metal_lb_ip_range }}
+{% if metal_lb_ip_range is string %}
```

Comment:
@timothystewart6 
> I know this might be a late ask but rather than change `metal_lb_ip_range` to an `array`.

> Just asking the question, doesn't mean it needs to change! 🙂

No worries, I'm also very much interested in finding a solid technical solution here 😊 Here are my thoughts on the matter:

 - If we were starting fresh, I would definitely recommend modelling this a list, for these reasons:
   - It is more flexible. I can imagine at least a few scenarios where one might want to specify multiple ranges even without dual stack being involved.
   - For the MetalLB CR, we transform it into a list anyway. So when modelling it as a list ourselves, it is easy to understand how we treat that list (just pass it along). This is one thing I do not like so much about the `metal_lb_ip_range_ipv6` proposal: It implies that IPv6 ranges would be somehow treated differently than IPv4 when they are not.
 - But we are not starting fresh and want to maintain backwards compatibility. So we have to make sure that we support the old format. I have two proposals here:
   - My preferred approach: Accept strings as well as lists. Test both cases. With molecule, we can do that quite easily. I will outline the concept below.
   - If we really want to make sure that the type remains string, we could go for comma-separated ranges, like so:
     ```yaml
     metal_lb_ip_range: >-
       192.0.2.10-20,
       192.0.2.55-64,
       2001:db8::1b:0/112
     ```
     For this we would not need a logic fork, but a little more complex logic for all cases.


## How to test with molecule

Molecule support is coming along quite nicely, I will probably open a PR in the next few days (superseding #60). Molecule supports [scenarios](https://molecule.readthedocs.io/en/latest/getting-started.html#molecule-scenarios), which are very useful here. We can have two scenarios:

- `default` uses the sample inventory (almost at least). A 3 + 2 node cluster with IPv4 only (since lots of people do not care about or understand IPv6 and nothing complicated should be going on in the sample inventory). We can model the IP ranges as a string in the sample host vars, like so:
  ```yaml
  # IP range for MetalLB load balancer
  # If you want to use multiple ranges, you can set this to a list of strings too.
  metal_lb_ip_range: "192.168.30.80-192.168.30.90"
  ```
- `ipv6`: This would be a dual-stack configuration (like I [described above](https://github.com/techno-tim/k3s-ansible/pull/48#issuecomment-1205794689). Additionally, to save some CI time and improve test coverage at the same time, this could be a non-HA cluster (my suggestion: 1 control + 1 worker node).

If you want to, we can first get molecule support done - and then I will update this PR with a matching molecule scenario so that the changes are well-tested from the start :-)

Reply:
Hey, thank you for the detailed explanation!  I am totally fine with the type change if we cast it properly (like you are) and then test both cases!  Thank you!

Reply:
If you don't mind then, let's hold off on this merge until the function tests with molecule are done!

Reply:
Test are now added 👍🏻